implementations/contracts/ERC725XCore.sol

```diff
@@ -35,24 +37,27 @@ abstract contract ERC725XCore is OwnableUnset, ERC165Storage, IERC725X {
         uint256 _value,
         bytes calldata _data
     ) public payable virtual override onlyOwner returns (bytes memory result) {
-        uint256 txGas = gasleft();
+        require(address(this).balance >= _value, "ERC725X: insufficient balance for call");
 
-        // prettier-ignore
+        uint256 txGas = gasleft();
```

Comment:
I am not totally sure, and it would be good to test.
My assumption is that it could be good to leave it, as since it forwards only a specific amount of gas, it gives a threshold of how much gas can be consumed.

This could prevent from reaching really high gas amounts (or going over the block gas limit in worst case), so maybe this is good for re-entrancy?

I am not totally sure, it would be good to test it.